avm/res/insights/data-collection-rule/main.bicep

```
@@ -0,0 +1,169 @@
+metadata name = 'Data Collection Rules'
+metadata description = 'This module deploys a Data Collection Rule.'
+metadata owner = 'Azure/module-maintainers'
+
+// ============== //
+//   Parameters   //
+// ============== //
+
+@sys.description('Required. The name of the data collection rule. The name is case insensitive.')
+param name string
+
+@sys.description('Optional. The resource ID of the data collection endpoint that this rule can be used with.')
+param dataCollectionEndpointId string = ''
+
+@sys.description('Required. The specification of data flows.')
+param dataFlows array
+
+@sys.description('Required. Specification of data sources that will be collected.')
+param dataSources object
+
+@sys.description('Optional. Description of the data collection rule.')
+param description string = ''
+
+@sys.description('Required. Specification of destinations that can be used in data flows.')
+param destinations object
+
+@sys.description('Optional. Enable/Disable usage telemetry for module.')
+param enableTelemetry bool = true
+
+@sys.description('Optional. The kind of the resource.')
+@allowed([
+  'Linux'
+  'Windows'
+])
+param kind string = 'Linux'
+
+@sys.description('Optional. Location for all Resources.')
+param location string = resourceGroup().location
+
+@sys.description('Optional. The lock settings of the service.')
+param lock lockType
+
+@sys.description('Optional. Array of role assignment objects that contain the \'roleDefinitionIdOrName\' and \'principalId\' to define RBAC role assignments on this resource. In the roleDefinitionIdOrName attribute, you can provide either the display name of the role definition, or its fully qualified ID in the following format: \'/providers/Microsoft.Authorization/roleDefinitions/c2f4ef07-c644-48eb-af81-4b1b4947fb11\'.')
```

```suggestion
@description('Optional. Array of role assignments to create.')
```
Update to the new version of the interface for Role Assignment  [Link](https://azure.github.io/Azure-Verified-Modules/specs/shared/interfaces/#role-assignments). Note this will need a readme regeneration.